### DISCUSSION TITLE
ESP32 WROOM, urequests Problem

### DISCUSSION THREAD

**Discussion by aiko929** [micropython/micropython#10568] [ESP32]
On my ESP32 im currently running micropython v1.19 with vscode and pymakr. 

My Code looks like this:

```
from machine import Pin
from time import sleep
import network
import urequests
import ujson

led_buildin = Pin(2, Pin.OUT)
url = "http://myurl:8080/get/info/id"
# Connect to Wifi:
wlan = network.WLAN(network.STA_IF) # create station interface
wlan.active(True)       # activate the interface

if not wlan.isconnected():
    wlan.connect("yyy", "xxx")

while wlan.isconnected():
    
    req = urequests.get(url)
    heizung_plug_status = ujson.loads(req.text)["status"]
    req.close()
    
    if heizung_plug_status:
        led_buildin.value(1)
    else:
        led_buildin.value(0)
    sleep(1)
```

The problem that i have is when im testing this in vscode everything runs fine and the build in led acts accordingly to what im returning through the response. However when i unplug the ESP32 from my PC and boot it up at the same location it doesnt create an response. ive tested with the build in led and it cant get past this line: 

`req = urequests.get(url)`

any ideas why it doesnt work standalone, but works just fine while connected to vscode?

**Comment by jimmo** [comment]
> However when i unplug the ESP32 from my PC and boot it up at the same location

How are you powering the ESP32 when it's not connected to the PC?

> `heizung_plug_status = ujson.loads(req.text)["status"]`

FWIW, you can write `req.json["status"]`. This will automatically `.close()` the request object too.

> ive tested with the build in led and it cant get past this line:

Can you clarify exactly what you mean here?

**Reply by aiko929** [reply]
Im powering it with a 5v 2a usb power supply from samsung.

thanks for this tip!

I tried to find the line where any error happens, so part by part i turned on the build_in led to see that i finished a line.
This way i found out that my program successfully runs until this line:

`req = urequests.get(url)`

Thanks for your fast answer!

**Comment by davefes** [comment]
Maybe the following might give more information:
```py
        try:
           req = urequests.get(url)
        except OSError as error:
            try:
                with open('errors.txt', 'a') as outfile:
                    outfile.write(str(error) + '\n')
            except OSError:
                pass
```

**Comment by aiko929** [comment]
Alright, the problem was somehow my VSCode Setup. ive switched to Thonny and everything worked as intended, thanks for helping!
